x-pack/plugins/security_solution/common/detection_engine/schemas/request/rule_schemas.ts

```diff
@@ -412,6 +415,14 @@ const responseRequiredFields = {
   updated_by,
   created_at,
   created_by,
+
+  // NOTE: For now, Related Integrations, Required Fields and Setup Guide are supported for prebuilt
+  // rules only. We don't want to allow users to edit these 3 fields via the API. If we added them
+  // to baseParams.defaultable, they would become a part of the request schema as optional fields.
+  // This is why we add them here, in order to add them only to the response schema.
```

Comment:
Appreciate the context here again. Hopefully when we refactor these schemas we can either have general comments or it'll be simple enough to make this obvious.